### PR TITLE
Serverless Cosmos exploration + cleanup [VS-906]

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,3 +89,73 @@ be much faster than the VS-890 baseline). The updated indexing policy should loo
...
+
+# VS-906 Cosmos DB Serverless Exploration
+
+[Cosmos DB serverless](https://learn.microsoft.com/en-us/azure/cosmos-db/serverless) is restricted to [50 GB per
+container and 5000 RU/s throughput](https://learn.microsoft.com/en-us/azure/cosmos-db/concepts-limits#serverless). There is a
+[preview 1 TB container](https://learn.microsoft.com/en-us/azure/cosmos-db/serverless-1tb) offering in which I have
+enrolled the Variants subscription which will offer higher throughput as storage grows.
+
+An ingest run with Quickstart data using the code from this spike consumed 6.65 M RU for reference data and 10.36
+M RU for variant data. Run times were ~32 minutes and ~37 for a total of ~69 minutes. RU consumption cost was
```
Comment (gbggrant) [inline README.md]:
```suggestion
M RU for variant data. Run times were ~32 minutes and ~37 minutes, respectively, for a total of ~69 minutes. RU consumption cost was
```

File: src/main/java/org/broadinstitute/gvs/azure/cosmos/AvroReader.java [hunk 2]
```diff
@@ -53,80 +54,138 @@ static long calculateEndLocation(ObjectNode record) {
...
-        long currentMaxLocation = -1L;
         List<ObjectNode> documentList = new ArrayList<>();
         ObjectNode currentDocument = null;
-        ArrayNode currentArrayNode = null;
+        ArrayNode currentRecordArray = null;
         Long currentSampleId = null;
+        long currentMaxLocation = -1L;
+        short currentChromosome = -1;
+        ArrayNode avroSchema = null;
+        // Why is this not a number in BQ?
```
Comment (gbggrant) [inline src/main/java/org/broadinstitute/gvs/azure/cosmos/AvroReader.java]: What is this comment referring to?

**Review by gbggrant — APPROVED** [review]

File: src/main/java/org/broadinstitute/gvs/azure/cosmos/AvroReader.java (hunk 2 above)
Comment (mcovarr) [reply src/main/java/org/broadinstitute/gvs/azure/cosmos/AvroReader.java]: Drop state on the next line. However Rori answered this question in mobbing (so we can store weird states that are not Phred-scaled qualities using letters), so this comment can go.

**Review by rsasch — APPROVED** [review]
LGTM

**Pull request merged**